packages/swap/src/routes/v2/swap.ts

```diff
@@ -0,0 +1,323 @@
+        refundEgress,
+        refundEgress?.broadcast,
+        ignoredEgresses,
+        'REFUND',
+        refundEgressTrackerTxRef,
+      ),
+      srcAsset && destAsset && estimateSwapDuration({ srcAsset, destAsset }),
+      internalSrcAsset && getRequiredBlockConfirmations(internalSrcAsset),
+    ]);
+
+    const response = {
+      state,
+      swapId: swapRequest?.nativeId.toString(),
+      ...(internalSrcAsset && getAssetAndChain(internalSrcAsset, 'src')),
+      ...(internalDestAsset && getAssetAndChain(internalDestAsset, 'dest')),
+      destAddress: readField(swapRequest, swapDepositChannel, failedSwap, 'destAddress'),
+      depositChannel: {
+        createdAt: swapDepositChannel?.createdAt.valueOf(),
+        brokerCommissionBps: swapDepositChannel?.brokerCommissionBps,
+        depositAddress: swapDepositChannel?.depositAddress,
+        expiryBlock: swapDepositChannel?.srcChainExpiryBlock?.toString(),
+        estimatedExpiryTime: swapDepositChannel?.estimatedExpiryAt?.valueOf(),
+        expectedDepositAmount: swapDepositChannel?.expectedDepositAmount?.toFixed(),
+        isExpired: swapDepositChannel?.isExpired,
+        openedThroughBackend: swapDepositChannel?.openedThroughBackend,
+        affiliateBrokers,
+        fillOrKillParams: swapDepositChannel?.fokMinPriceX128
+          ? {
+              retryDurationBlocks: swapDepositChannel.fokRetryDurationBlocks,
+              refundAddress: swapDepositChannel.fokRefundAddress,
+              minPrice: getPriceFromPriceX128(
+                swapDepositChannel.fokMinPriceX128.toFixed(),
+                swapDepositChannel.srcAsset,
+                swapDepositChannel.destAsset,
+              ),
+            }
+          : undefined,
+        dcaParams: swapDepositChannel?.chunkIntervalBlocks
+          ? {
+              numberOfChunks: swapDepositChannel?.numberOfChunks,
+              chunkIntervalBlocks: swapDepositChannel?.chunkIntervalBlocks,
+            }
+          : undefined,
+      },
+      deposit: {
+        amount:
+          readField(swapRequest, failedSwap, 'depositAmount')?.toFixed() ?? pendingDeposit?.amount,
+        transactionRef: depositTransactionRef,
+        txConfirmations: pendingDeposit?.transactionConfirmations,
+        receivedAt: swapRequest?.depositReceivedAt?.valueOf(),
+        receivedBlockIndex: swapRequest?.depositReceivedBlockIndex ?? undefined,
+        ...(failedSwap && { failure: getDepositIgnoredFailedState(failedSwap) }),
+      },
+      swap: {
+        ...rolledSwaps,
+        totalAmountSwapped: rolledSwaps?.totalAmountSwapped.toFixed(),
+        lastExecutedChunk:
+          rolledSwaps?.lastExecutedChunk && getSwapFields(rolledSwaps.lastExecutedChunk),
+        currentChunk: rolledSwaps && getSwapFields(rolledSwaps.currentChunk),
+        ...swapEgressFields,
+        type: sortedSwaps?.[0].type,
+        fees: aggregateFees,
+        srcChainRequiredBlockConfirmations,
+        estimatedDurationSeconds,
```

Comment:
this includes the witness time for the deposit - so i think we should move it to the root level